packages/application-extension/src/index.tsx

```diff
@@ -170,15 +168,13 @@ const mainCommands: JupyterFrontEndPlugin<void> = {
     ): DockLayout.ITabAreaConfig | null => {
       switch (area.type) {
         case 'split-area': {
-          const iterator = iter(area.children);
+          const it = area.children[Symbol.iterator]();
           let tab: DockLayout.ITabAreaConfig | null = null;
-          let value: DockLayout.AreaConfig | undefined;
-          do {
-            value = iterator.next();
-            if (value) {
-              tab = findTab(value, widget);
-            }
-          } while (!tab && value);
+          let item = it.next();
+          while (!tab && !item.done) {
+            tab = findTab(item.value, widget);
+            item = it.next();
+          }
```

Comment:
Could we use `find` to ease readability?

Reply:
That's a good idea. This is a recursive function that should probably be refactored. I think I'd rather put that in a different PR, though, if you're okay with that.